plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -765,6 +766,74 @@ public void testCollectStatsAfterColumnAdded(boolean collectOnWrite)
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testForceRecalculateStatsWithDeleteAndUpdate()
+    {
+        String tableName = "test_analyze_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName
+                + " AS SELECT * FROM tpch.sf1.nation", 25);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, 25.0, 0.0, null, 0, 24)," +
+                        "('regionkey', null, 5.0, 0.0, null, 0, 4)," +
+                        "('comment', 1857.0, 25.0, 0.0, null, null, null)," +
+                        "('name', 177.0, 25.0, 0.0, null, null, null)," +
+                        "(null, null, null, null, 25.0, null, null)");
+
+        // check that analyze does not change already calculated statistics
+        assertUpdate("ANALYZE " + tableName);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, 25.0, 0.0, null, 0, 24)," +
+                        "('regionkey', null, 5.0, 0.0, null, 0, 4)," +
+                        "('comment', 1857.0, 25.0, 0.0, null, null, null)," +
+                        "('name', 177.0, 25.0, 0.0, null, null, null)," +
+                        "(null, null, null, null, 25.0, null, null)");
+
+        assertUpdate("DELETE FROM " + tableName + " WHERE nationkey = 1", 1);
```

Comment:
It is not very clear from the test why the table stats are not updated when doing `UPDATE` / `DELETE`. Could you please add a comment in this regard?

Reply:
They are updated but not correctly, there is a difference in expected values, I am not 100% sure about the reason of this.

Reply:
yes, but I wanted them to change so I can show the diff later, i hope that it makes sense